viewing_party/party.py

```diff
@@ -1,23 +1,119 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    '''
+    Create a dict containing info about one movie, with "title", "genre", and "rating", as keys, mapped to values passed in as arguments.
+    '''
+    if not title or not genre or not rating:
+        return None
 
+    movie = {}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    '''
+    Add a movie to a user's list of watched movies
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    Add a movie to a user's watchlist
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    '''
+    Move a movie from watchlist to watched
+    '''
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            watched_movie = movie
+            user_data["watched"].append(watched_movie)
+            user_data["watchlist"].remove(watched_movie)
+    
+    return user_data
+    
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    ratings = []
+    for movie in user_data["watched"]:
+        ratings.append(movie["rating"])
+    if not ratings:
+        return 0
+    
+    return sum(ratings)/len(ratings)
 
-
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+    genres = {}
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genres:
+            genres[movie["genre"]] = 1
+        else:
+            genres[movie["genre"]] += 1
+    
+    return max(genres, key = genres.get)

```

Comment:
Great use of a frequency dictionary! The last line is cool and works, but as practice consider how you would do it in a more manual way without using `max`.